helpers/decrementVar.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+const utils = require('handlebars-utils');
+const max_keys = 50;
+
+const factory = globals => {
+    return function(key) {
+        if (!utils.isString(key)) {
+            throw new Error("decrementVar helper key must be a string");
+        }
+
+        // Setup storage
+        if (typeof globals.storage.variables === 'undefined') {
+            globals.storage.variables = {};
+        }
+
+        // Make sure the number of total keys is within the limit
+        if (Object.keys(globals.storage.variables).length >= max_keys) {
```

Comment:
You should probably move this down into the conditional since we only need this guard if creating a new variable.